### PR TITLE
Add CI configuration to project.

### PR DESCRIPTION
**Pull request by czyszczonik**
#3

### CODE REVIEW

**Comment by czyszczonik** [comment]
You [@mbore] must add this project in your circleci app, because you are owner. And activate option: run ci only for pull requests

File: .circleci/config.yml [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+      - image: circleci/openjdk:8u242-jdk
+    steps:
+      - checkout
+      - restore_cache:
+          keys:
+            - gradle-{{ checksum "build.gradle" }}
+      - run: gradle wrapper
+      - run: ./gradlew dependencies
+      - run: ./gradlew test
+      - run: ./gradlew jar
```
Comment (jak-b) [inline .circleci/config.yml]: Give each run a name for better readability. See: https://circleci.com/docs/2.0/configuration-reference/#run

File: .circleci/config.yml [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+          destination: libs
+  data:
+    docker:
+      - image: circleci/openjdk:8u242-jdk
+    steps:
+      - checkout
+      - run: echo TODO
+      - store_artifacts:
+          path: FROM
+          destination: TO
```
Comment (jak-b) [inline .circleci/config.yml]: What is the purpose of this job?

File: .circleci/config.yml (hunk 1 above)
Comment (jak-b) [inline .circleci/config.yml]: Do we have to build jar file at this point?

File: .circleci/config.yml [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+      - run: ./gradlew dependencies
+      - run: ./gradlew test
+      - run: ./gradlew jar
+      - save_cache:
+          paths:
+            - ~/.gradle
+          key: gradle-{{ checksum "build.gradle" }}
+      - store_artifacts:
+          path: application/build/libs
+          destination: libs
```
Comment (jak-b) [inline .circleci/config.yml]: What is the purpose of storing a build?

File: .circleci/config.yml [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+            - gradle-{{ checksum "build.gradle" }}
+      - run: gradle wrapper
+      - run: ./gradlew dependencies
+      - run: ./gradlew test
+      - run: ./gradlew jar
+      - save_cache:
+          paths:
+            - ~/.gradle
+          key: gradle-{{ checksum "build.gradle" }}
+      - store_artifacts:
```
Comment (jak-b) [inline .circleci/config.yml]: You could consider storing test results for display in test summary. See https://circleci.com/docs/2.0/collect-test-data/

File: .circleci/config.yml [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+      - restore_cache:
+          keys:
+            - gradle-{{ checksum "build.gradle" }}
+      - run: gradle wrapper
+      - run: ./gradlew dependencies
+      - run: ./gradlew test
+      - run: ./gradlew jar
+      - save_cache:
+          paths:
+            - ~/.gradle
```
Comment (jak-b) [inline .circleci/config.yml]: Do we have to cache the whole `.gradle` directory? In my opinion storing `.gradle/wrapper` and `.gradle/caches` is sufficient. Also I would store them separately. Something like:
```
    - save_cache:
          paths:
            - ~/.gradle/wrapper
          key: gradle-wrapper-{{ checksum "gradle/wrapper/gradle-wrapper.properties" }}
      - save_cache:
          paths:
            - ~/.gradle/caches
          key:gradle-cache-{{ checksum "build.gradle" }}
```

File: .circleci/config.yml [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+  build:
+    docker:
+      - image: circleci/openjdk:8u242-jdk
+    steps:
+      - checkout
+      - restore_cache:
+          keys:
+            - gradle-{{ checksum "build.gradle" }}
+      - run: gradle wrapper
+      - run: ./gradlew dependencies
```
Comment (jak-b) [inline .circleci/config.yml]: Why? From gradle help: `dependencies - Displays all dependencies declared in root project 'pns'.`

File: .circleci/config.yml [hunk 7]
```diff
@@ -0,0 +1,44 @@
...
+jobs:
+  build:
+    docker:
+      - image: circleci/openjdk:8u242-jdk
+    steps:
+      - checkout
+      - restore_cache:
+          keys:
+            - gradle-{{ checksum "build.gradle" }}
+      - run: gradle wrapper
```
Comment (jak-b) [inline .circleci/config.yml]: I think that we should add `gradlew` and `gradle.bat` files to the repository. I this case this step could be omitted. See https://docs.gradle.org/current/userguide/gradle_wrapper.html#sec:adding_wrapper

File: .circleci/config.yml (hunk 5 above)
Comment (mbore) [reply .circleci/config.yml]: Why would you like to store them separately?

File: .circleci/config.yml [hunk 8]
```diff
@@ -0,0 +1,44 @@
...
+      - run: ./gradlew dependencies
+      - run: ./gradlew test
```
[the remark this answers is not in the document]
Comment (czyszczonik) [reply .circleci/config.yml]: Done!

File: .circleci/config.yml (hunk 4 above)
Comment (czyszczonik) [reply .circleci/config.yml]: Done!

File: .circleci/config.yml (hunk 5 above)
Comment (jak-b) [reply .circleci/config.yml]: It's ok if there is a possibility to store them together. I just wanted not to include the whole `.gradle` directory.

**Pull request merged**